superset/assets/src/chart/chartAction.js

```diff
@@ -202,5 +202,10 @@ export function runQuery(formData, force = false, timeout = 60, key) {
 }
 
 export function refreshChart(chart, force, timeout) {
-  return dispatch => dispatch(runQuery(chart.latestQueryFormData, force, timeout, chart.id));
+  return dispatch => {
+    if (Object.keys(chart.latestQueryFormData).length === 0) {
```

Comment:
should there also be a check for whether `chart.latestQueryFormData` is defined?

Reply:
it is getInitialState in line 79:
https://github.com/apache/incubator-superset/blob/c06531950869d84a50e7914b945ba6ceb2a5c179/superset/assets/src/dashboard/reducers/getInitialState.js#L79

Reply:
can we add the check JIC since there is no downside to adding it?

Reply:
added.